packages/truffle-compile/compilerSupplier/loadingStrategies/LoadingStrategy.js

```diff
@@ -15,11 +15,14 @@ class LoadingStrategy {
     };
     this.config = Object.assign({}, defaultConfig, options);
 
-    const compilerCachePath = path.resolve(
+    const compilersDir = path.resolve(
       Config.getTruffleDataDirectory(),
       "compilers"
     );
-    if (!fs.existsSync(compilerCachePath)) fs.mkdirSync(compilerCachePath);
+    const compilerCachePath = path.resolve(compilersDir, "node_modules"); // because babel binds to require & does weird things
+    if (!fs.existsSync(compilersDir)) fs.mkdirSync(compilersDir);
+    if (!fs.existsSync(compilerCachePath)) fs.mkdirSync(compilerCachePath); // for 5.0.8 users
```

Comment:
Could you explain more about the reasoning behind this case? Not sure about what "for 5.0.8 users" means.

Reply:
5.0.8 users now have a `compilers` directory in their truffle config path if they've used BYOC.

Tyler brought up the good point of making sure the original PR logic didn't skip over them (it was a check just for `compilers` before creating `compilers` & then `node_modules`)